CMSIS/Core/Include/r-profile/armv7r.h

```diff
@@ -0,0 +1,471 @@
+/**************************************************************************//**
+ * @file     armv7r.h
+ * @brief    CMSIS Cortex-R Core Peripheral Access Layer Header File for ARMv7-R
+ * @version  V1.0.0
+ * @date     2. July 2023
+ ******************************************************************************/
+/*
+ * Copyright (c) 2009-2023 ARM Limited. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if   defined ( __ICCARM__ )
+  #pragma system_include         /* treat file as system include file for MISRA check */
+#elif defined (__clang__)
+  #pragma clang system_header   /* treat file as system include file */
+#endif
+
+#ifndef __ARM_V7R_GENERIC
+#define __ARM_V7R_GENERIC
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/*******************************************************************************
+ *                 CMSIS definitions
+ ******************************************************************************/
+ /**
+  \ingroup ARMv7-R
+  @{
+ */
+
+#include "cmsis_version.h"
+
+#if defined ( __CC_ARM )
+  #if defined (__TARGET_FPU_VFP)
+    #if defined (__FPU_PRESENT) && (__FPU_PRESENT == 1U)
+      #define __FPU_USED       1U
+    #else
+      #warning "Compiler generates FPU instructions for a device without an FPU (check __FPU_PRESENT)"
```

Comment:
replace `#warning` with `#error`. GCC does not support `#warning`.

Reply:
GCC does support #warning, however it is con compliant to ANSI-C nor to C++.